src/core/hle/kernel/k_server_port.h

```diff
@@ -0,0 +1,80 @@
+// Copyright 2021 yuzu emulator team
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <boost/intrusive/list.hpp>
+
+#include "common/common_types.h"
+#include "core/hle/kernel/k_server_session.h"
+#include "core/hle/kernel/k_synchronization_object.h"
+#include "core/hle/result.h"
+
+namespace Kernel {
+
+class KernelCore;
+class KPort;
+class SessionRequestHandler;
+
+class KServerPort final : public KSynchronizationObject {
+    KERNEL_AUTOOBJECT_TRAITS(KServerPort, KSynchronizationObject);
+
+private:
+    using SessionList = boost::intrusive::list<KServerSession>;
+
+public:
+    explicit KServerPort(KernelCore& kernel);
+    virtual ~KServerPort() override;
+
+    using HLEHandler = std::shared_ptr<SessionRequestHandler>;
+
+    void Initialize(KPort* parent_, std::string&& name_);
+
+    /// Whether or not this server port has an HLE handler available.
+    bool HasHLEHandler() const {
+        return hle_handler != nullptr;
+    }
+
+    /// Gets the HLE handler for this port.
+    HLEHandler GetHLEHandler() const {
+        return hle_handler;
+    }
+
+    /**
+     * Sets the HLE handler template for the port. ServerSessions crated by connecting to this port
+     * will inherit a reference to this handler.
+     */
+    void SetHleHandler(HLEHandler hle_handler_) {
+        hle_handler = std::move(hle_handler_);
+    }
+
+    void EnqueueSession(KServerSession* pending_session);
+
+    KServerSession* AcceptSession();
+
+    constexpr const KPort* GetParent() const {
```

Comment:
```suggestion
    const KPort* GetParent() const {
```

This can't be used in a constexpr context based off the interface